### ISSUE TITLE
test_basic.test_trapping_of_bad_request_key_errors failed in Travis-CI

### ISSUE THREAD

**Issue by greyli** [pallets/flask#3043]
[flask.tests.test_basic.test_trapping_of_bad_request_key_errors](https://github.com/pallets/flask/blob/master/tests/test_basic.py#L1029-L1040) failed in Travis-CI with enviroment `devel`.

### Source

[`flask.tests.test_basic.test_trapping_of_bad_request_key_errors`](https://github.com/pallets/flask/blob/master/tests/test_basic.py#L1029-L1040)

```py
def test_trapping_of_bad_request_key_errors(app, client):
    @app.route('/key')
    def fail():
        flask.request.form['missing_key']

    @app.route('/abort')
    def allow_abort():
        flask.abort(400)

    rv = client.get('/key')
    assert rv.status_code == 400
    assert b'missing_key' not in rv.data  # unpassed
```

### Traceback

https://travis-ci.org/pallets/flask/builds/471002531

```pytb
devel create: /home/travis/build/pallets/flask/.tox/devel
devel installdeps: pytest>=3, coverage, greenlet, blinker, python-dotenv, 
https://github.com/pallets/werkzeug/archive/master.tar.gz, 
https://github.com/pallets/markupsafe/archive/master.tar.gz, 
https://github.com/pallets/jinja/archive/master.tar.gz, 
https://github.com/pallets/itsdangerous/archive/master.tar.gz, 
https://github.com/pallets/click/archive/master.tar.gz
devel inst: /home/travis/build/pallets/flask/.tox/.tmp/package/1/Flask-1.0.3.dev0.zip
...
___________________ test_trapping_of_bad_request_key_errors ____________________
app = <Flask 'flask_test'>, client = <FlaskClient <Flask 'flask_test'>>
    def test_trapping_of_bad_request_key_errors(app, client):
        @app.route('/key')
        def fail():
            flask.request.form['missing_key']
    
        @app.route('/abort')
        def allow_abort():
            flask.abort(400)
    
        rv = client.get('/key')
        assert rv.status_code == 400
>       assert b'missing_key' not in rv.data
E       assert 'missing_key' not in "<!DOCTYPE HTML PUBL...'</code></pre></p>\n"
E         'missing_key' is contained here:
E           yError: \'missing_key\'</code></pre></p>
E         ?           +++++++++++
```

**Comment by garenchan** [comment]
This seems to be caused by `werkzeug`'s modification.
https://github.com/pallets/werkzeug/commit/da9d70a33

@davidism, I don't think it's appropriate to feed back these error messages to the user in non-debug mode.

https://github.com/pallets/werkzeug/blob/eda35e52418fc86689320b1d27c9f740f7d0aed1/werkzeug/exceptions.py#L114-L123
```phthon
            def get_description(self, environ=None):
                out = super(cls, self).get_description(environ=environ)

                if self.args:
                    out += "<p><pre><code>{}: {}</code></pre></p>".format(
                        exception.__name__,
                        escape(exception.__str__(self))
                    )

                return out
```

**Comment by davidism** [comment]
Fixed in #3063
